test/extended/images/mongodb_replica.go

```diff
@@ -0,0 +1,126 @@
+package images
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	"github.com/openshift/origin/test/extended/util/db"
+)
+
+var _ = g.Describe("[images][mongodb] openshift mongodb replication", func() {
+	defer g.GinkgoRecover()
+
+	const (
+		templatePath         = "https://raw.githubusercontent.com/openshift/mongodb/master/2.4/examples/replica/mongodb-clustered.json"
+		deploymentConfigName = "mongodb"
+		insertCmd            = `db.bar.save({ "status": "passed" })`
+	)
+
+	const (
+		expectedReplicasAfterDeployment = 3
+		expectedReplicasAfterScalingUp  = expectedReplicasAfterDeployment + 2
+	)
+
+	oc := exutil.NewCLI("mongodb-replica", exutil.KubeConfigPath()).Verbose()
+
+	g.Describe("creating from a template", func() {
```

Comment:
@rhcarvalho @bparees I did small experiments and I couldn't to refactor code to the suggested structure:
1) if we will use 3 `Describe` blocks then we lost the order, all these block will be run in random order. But we can't read/write or scale up until service will be created
2) if we will use 3 `It` blocks then we still need to have a shared variable between 3 blocks. In the 2nd and 3rd blocks we need to have a list of pods at least. When I extracted shared variable I got the `invalid memory address or nil pointer dereference` error. I didn't try more options because I want to reduce amount of time spending on this task.